src/framework/mpas_dmpar.F

```diff
@@ -5458,6 +5468,9 @@ subroutine mpas_dmpar_exch_halo_field2d_real(field, haloLayersIn)!{{{
           end do
         else
           nHaloLayers = size(field % sendList % halos)
+#ifdef MPAS_DEBUG
+          call mpas_log_write('exch_halo nHaloLayers:$i destList halos:$i', intArgs=(/nHaloLayers, size(field%recvList%halos)/))
```

Comment:
Rather than using a direct call to `mpas_log_write` guarded by a check on the `MPAS_DEBUG` macro definition, I think the preferred way to add a debugging print statement would be with the `DMPAR_DEBUG_WRITE` macro; e.g.,
```
          DMPAR_DEBUG_WRITE('exch_halo nHaloLayers:$i destList halos:$i' COMMA intArgs=(/nHaloLayers COMMA size(field%recvList%halos)/))
```